test/feed/index.spec.ts

```
@@ -18,6 +20,18 @@ async function uploadChunk(url: string, index: number): Promise<ChunkReference>
   return hexToBytes(referenceResponse.reference as HexString) as ChunkReference
 }
 
+async function tryUploadFeedUpdate(url: string, signer: Signer, topic: Topic, index: Index, reference: ChunkReference) {
```

This seems a bit wrong to me. Prior to running the test we should set up a valid state and then we should not ignore any errors (eq. to "try"), because the outcome should be correct or there is some problem and then it should fail.

---

This function is only supposed to be used for preparing the node state for finding feed updates and not for testing feed updates. I tried to emphasize it with the naming, maybe adding a comment would make it more clear?

---

Well, my problem is, what will happen if the update won't happen? Eq. there was already some SOC chunk on the given index, so the referenced CAC won't get updated. Then the test might fail for some assertions on what is expected to be referenced...

Generally, this approach depends on to state prior to running the test case which in my experience is a bad thing. Every test should be as isolated as possible. For example the order of tests execution might change and then suddenly there might be unexpected failures, or you run only limited set of test cases, which then might lead to same problem, or you add new test later on that might upload SOC to that address and introduce conflicts...

---

This workaround is only for make the tests repeatable in a local environment without the need to stop the nodes, delete the data and restart the nodes. On CI we have a new, clean environment all the time. This guarantees that separate test runs are reproducible. It's also possible to do this with the provided `/test/bee.sh` script using the `--ephemeral` switch, but it's inconvenient to do so. We also considered having a debug endpoint to delete all data from a node that could be invoked before running our tests but that's also not trivial to do because of the nature of a decentralized push synching that the Bee nodes do.

https://github.com/ethersphere/bee/issues/1264

Also please note that these tests are independent of each other because they use different topics. The function in question is actually used to help setting up a test where we are checking if the lookup works for multiple updates, so it's not that important what the uploaded content is. Of course it's possible to write data to your local test cluster that will make tests fail. In that case (or if you are in doubt regarding your tests) you can always stop the test nodes, delete all data, restart and see if the problem persist.

---

I also don't like this solution long term and would love to have some `// FIXME: ` note here and probably issue about it with `need help` label.

---

Done, see https://github.com/ethersphere/bee-js/issues/154